super-productivity: init at 1.10.45

offlinehacker opened the pull request.

offlinehacker committed:
super-productivity: init at 1.10.45

GrahamcOfBorg commented:
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43142&attempt_id=37adf5b7-9120-43b5-a4fe-bf8757b1da2c)

The following builds were skipped because they don't evaluate on aarch64-linux: super-productivity

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43142&attempt_id=73d11668-4008-460f-8403-f1a5388aae6e)

Attempted: super-productivity

<details><summary>Partial log (click to expand)</summary><p>

```

trying https://github.com/johannesjo/super-productivity/releases/download/1.10.45/superProductivity_1.10.45_amd64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404 Not Found
error: cannot download superProductivity_1.10.45_amd64.deb from any mirror
builder for '/nix/store/4lm8lrzlg9dvkk51p7q5lqq14xkac9pz-superProductivity_1.10.45_amd64.deb.drv' failed with exit code 1
cannot build derivation '/nix/store/skyvm4mk60cmgwffx7svi084py122jni-super-productivity-1.10.45.drv': 1 dependencies couldn't be built
error: build of '/nix/store/skyvm4mk60cmgwffx7svi084py122jni-super-productivity-1.10.45.drv' failed
```
</p></details>

GrahamcOfBorg commented:
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43142&attempt_id=71a513fd-33b6-464c-b0d2-cd3d0a181c93)

The following builds were skipped because they don't evaluate on aarch64-linux: super-productivity

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43142&attempt_id=49843d64-c105-47f0-ac86-6a4daabc670d)

Attempted: super-productivity

<details><summary>Partial log (click to expand)</summary><p>

```
copying path '/nix/store/whk2gs9h5ksykqbq33vw9rd24c7bgiqy-xprintidle-ng-git-2015-09-01' from 'https://cache.nixos.org'...
building '/nix/store/7dzav7hv96c9g1jxmq77wa1jkcpxvkzh-superProductivity_1.10.45_amd64.deb.drv'...

trying https://github.com/johannesjo/super-productivity/releases/download/v1.10.45/superProductivity_1.10.45_amd64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   625    0   625    0     0   1636      0 --:--:-- --:--:-- --:--:--  1636
100 38.0M  100 38.0M    0     0  10.5M      0  0:00:03  0:00:03 --:--:-- 12.8M
building '/nix/store/agmi4knksw4wmypd27q6csxwrz0hda3m-super-productivity-1.10.45.drv'...
/nix/store/5dvqqmn61qpkyr6bywdcf8wsfgy7jnzi-super-productivity-1.10.45
```
</p></details>

adisbladis commented on pkgs/applications/networking/super-productivity/default.nix:
```diff
@@ -0,0 +1,101 @@
+  nativeBuildInputs = [ makeWrapper ];
+
+  unpackPhase = "true";
+  buildCommand = ''
```
It's a bit awkward to have all phases in one like this, makes overriding difficult.

offlinehacker replied:
I just copied slack derivation, i can put it in install phase nad disable other phases, do you think that would be better?

adisbladis commented on pkgs/applications/networking/super-productivity/default.nix:
```diff
@@ -0,0 +1,101 @@
+      --replace /opt/superProductivity/ $out/bin/
+  '';
+
+  meta = with stdenv.lib; {
```
No maintainer?

offlinehacker replied:
Well, again, slack package did not have maintainer, so i forgot.

offlinehacker commented:
Ok, i split into `unpackPhase` and `installPhase`, i also called `preInstall` and `postInstall` hooks from `installPhase`. For `fixupPhase` i had to disable striping. Also added me as maintainer.

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43142&attempt_id=d0329861-5813-426b-8baf-436c5960e998)

Attempted: super-productivity

<details><summary>Partial log (click to expand)</summary><p>

```
glibPreFixupPhase
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/lwhr13k02sksh0n3mdf4vyllppns9i9r-super-productivity-1.10.45
shrinking /nix/store/lwhr13k02sksh0n3mdf4vyllppns9i9r-super-productivity-1.10.45/libexec/superProductivity/libnode.so
shrinking /nix/store/lwhr13k02sksh0n3mdf4vyllppns9i9r-super-productivity-1.10.45/libexec/superProductivity/superproductivity
shrinking /nix/store/lwhr13k02sksh0n3mdf4vyllppns9i9r-super-productivity-1.10.45/libexec/superProductivity/libffmpeg.so
patching script interpreter paths in /nix/store/lwhr13k02sksh0n3mdf4vyllppns9i9r-super-productivity-1.10.45
head: error writing 'standard output': Broken pipe
checking for references to /build in /nix/store/lwhr13k02sksh0n3mdf4vyllppns9i9r-super-productivity-1.10.45...
/nix/store/lwhr13k02sksh0n3mdf4vyllppns9i9r-super-productivity-1.10.45
```
</p></details>

GrahamcOfBorg commented:
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43142&attempt_id=ac14b2fb-fdcd-41d3-b3ea-f8fc6372d750)

The following builds were skipped because they don't evaluate on aarch64-linux: super-productivity

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

adisbladis commented:
Much better. Thanks :)
